packages/core/src/new-api/internal/execution/execution-state-reducer.ts

```diff
@@ -145,6 +149,7 @@ export function executionStateReducer(
     const updateWithOnchainAction: ExecutionState = {
       ...previousExState,
       history: [...previousExState.history, action],
+      onchain: onchainActionReducer(previousExState.onchain, action),
```

Comment:
We build up per onchain action state under this property - open to new name suggestions!